22.md

```diff
@@ -0,0 +1,193 @@
+
+`q` tags MAY be used when citing events in the `.content` with [NIP-21](21.md).
+
+```json
+["q", "<event-id> or <event-address>", "<relay-url>", "<pubkey-if-a-regular-event>"]
+```
+
+`p` tags SHOULD be used when mentioning pubkeys in the `.content` with [NIP-21](21.md).
+If the parent item is an event, a `p` tag set to the parent event's author SHOULD be added.
+
+```json
+["p", "<pubkey>"]
+```
+
+## Examples
+
+A comment on a blog post looks like this:
+
+```js
+{
+  kind: 1111,
+  content: 'Great blog post!',
+  tags: [
+    // top-level comments scope to event addresses or ids
+    ["A", "30023:3c9849383bdea883b0bd16fece1ed36d37e37cdde3ce43b17ea4e9192ec11289:f9347ca7", "wss://example.relay"],
+    // the root kind
+    ["K", "30023"],
+
+    // the parent event address (same as root for top-level comments)
+    ["a", "30023:3c9849383bdea883b0bd16fece1ed36d37e37cdde3ce43b17ea4e9192ec11289:f9347ca7", "wss://example.relay"],
+    // when the parent event is replaceable or addressable, also include an `e` tag referencing its id
+    ["e", "5b4fc7fed15672fefe65d2426f67197b71ccc82aa0cc8a9e94f683eb78e07651", "wss://example.relay"],
+    // the parent event kind
+    ["k", "30023"]
+  ]
+  // other fields
+}
+```
+
+A comment on a [NIP-94](94.md) file looks like this:
+
+```js
+{
+  kind: 1111,
+  content: 'Great file!',
+  tags: [
+    // top-level comments have the same scope and reply to addresses or ids
+    ["E", "768ac8720cdeb59227cf95e98b66560ef03d8bc9a90d721779e76e68fb42f5e6", "wss://example.relay", "3721e07b079525289877c366ccab47112bdff3d1b44758ca333feb2dbbbbe5bb"],
+    // the root kind
+    ["K", "1063"],
+
+    // the parent event id (same as root for top-level comments)
+    ["e", "768ac8720cdeb59227cf95e98b66560ef03d8bc9a90d721779e76e68fb42f5e6", "wss://example.relay", "3721e07b079525289877c366ccab47112bdff3d1b44758ca333feb2dbbbbe5bb"],
+    // the parent kind
+    ["k", "1063"]
+  ]
+  // other fields
+}
+```
+
+A reply to a comment looks like this:
+
+```js
+{
+  kind: 1111,
+  content: 'This is a reply to "Great file!"',
+  tags: [
+    // nip-94 file event id
+    ["E", "768ac8720cdeb59227cf95e98b66560ef03d8bc9a90d721779e76e68fb42f5e6", "wss://example.relay", "fd913cd6fa9edb8405750cd02a8bbe16e158b8676c0e69fdc27436cc4a54cc9a"],
+    // the root kind
+    ["K", "1063"],
+
+    // the parent event
+    ["e", "5c83da77af1dec6d7289834998ad7aafbd9e2191396d75ec3cc27f5a77226f36", "wss://example.relay", "93ef2ebaaf9554661f33e79949007900bbc535d239a4c801c33a4d67d3e7f546"],
+    // the parent kind
+    ["k", "1111"]
+  ]
+  // other fields
+}
+```
+
+A comment on a website's url looks like this:
+
+```js
+{
+  kind: 1111,
+  content: 'Nice article!',
+  tags: [
+    // referencing the root url
+    ["S", "r:https://abc.com/articles/1"],
+    // the root "kind": for an url, the kind is its domain
+    ["K", "r:https://abc.com"],
+
+    // the parent reference (same as root for top-level comments)
+    ["s", "r:https://abc.com/articles/1"],
```

Comment:
Maybe when replying to the root, the small `s`/`k` can be omitted since its just a copy of the tags, or is there a use case for loading ONLY replies to root?

Reply:
I think the answer is yes